src/components/tasks/card/index.tsx

```diff
@@ -106,15 +108,22 @@ const Card: FC<Props> = ({
             height={iconHeight}
           />
           <span className={classNames.cardSpecialFont}>Due Date</span>
-          <span
-            className={classNames.cardStrongFont}
-            contentEditable={shouldEdit}
-            onKeyPress={(e) => handleChange(e, 'endsOn')}
-            role="button"
-            tabIndex={0}
-          >
-            {fromNowEndsOn}
-          </span>
+          {shouldEdit ? (
+            <input
+              type='date'
+              onChange={(e) => setDateTime(e.target.value)}
+              onKeyPress={(e) => handleChange(e, 'endsOn')}
+              value={dateTime}
+            />
+          ) : (
+            <span
+              className={classNames.cardStrongFont}
+              role='button'
+              tabIndex={0}
+            >
+              {fromNowEndsOn}
+            </span>
+          )}
```

Comment:
Nit: For simplicity, you can actually move this if-else logic into a function `renderDate(isEditing)`. It will make it easier to read this current component and abstract the logic of showing picker/span into it's own single responsibility function.

Reply:
Done!